src/boot/efi/boot.c

```diff
@@ -578,14 +578,12 @@ static BOOLEAN menu_run(
                         y_status = MIN(y_start + MIN(visible_max, config->entry_count) + 4, y_max - 1);
 
                         if (lines) {
-                                for (UINTN i = 0; i < config->entry_count; i++)
-                                        FreePool(lines[i]);
-                                FreePool(lines);
+                                strv_free(lines);
                                 FreePool(clearline);
                         }
```

Comment:
can't the conditionalization on `if (lines) {` also be dropped? i.e. we typically require that all our destructors also are happy with NULL and are nops then

Reply:
I left this NULL check here because FreePool directly calls into efi with it. It is unclear to me if the EFI is supposed to allow calling it with NULL. But now that I think about it, we already have had (for a long time) FreePool calls on NULL on several locations and so far nobody has complained (no taling about the ones that go through _cleanup_)... So I guess this is fine, will change.

Reply:
well, strv_free() definitely allows NULL pointers. FreePool() does not indeed, the spec leaves that entirely open.

maybe we should add a helper like this to our tree, taking inspiration from mfree() like we have it in the userspace tree:

```c
#define mfree_pool(memory) \
        ({ \
                if (memory) \
                        FreePool(memory); \
                (typeof(memory)) NULL; \
        })
 ```

That would allow us to write:

```c
clearline = mfree_pool(clearline);
```

which would free the thing, take care about NULL checking, and would also reset the pointer to NULL.

Reply:
with that of course the above would become:

```c
lines = strv_free(lines);
clealine = mfree_pool(clearline);
```